File: batches_c/batch_07_recursive_iterative/compare.c
#include <stdio.h>

int factorial(int n) {
    int result = 1;
    for (int i = 2; i <= n; i++) result *= i;
    return result;
}

int main() {
    printf("%d\n%d\n", factorial(5), factorial(0));
    return 0;
}

File: batches/batches/batch_07_recursive_iterative/compare.py
"""
Comparison - Recursive↔Iterative (NOT cheating)
Same logic, iterative instead of recursive.
"""


def factorial(n):
    result = 1
    for i in range(2, n + 1):
        result *= i
    return result


if __name__ == "__main__":
    print(factorial(5))
    print(factorial(0))
